src/sgame/sg_admin.cpp

```diff
@@ -3680,6 +3686,56 @@ bool G_admin_changemap( gentity_t *ent )
 	return true;
 }
 
+bool G_admin_changedevmap( gentity_t *ent )
+{
+	char map[ MAX_QPATH ];
+	char layout[ MAX_QPATH ] = { "" };
+
+	if ( trap_Argc() < 2 )
+	{
+		ADMP( QQ( N_("^3changedevmap:^* usage: changemap [map] (layout)") ) );
+		return false;
+	}
+
+	trap_Argv( 1, map, sizeof( map ) );
+
+	if ( !G_MapExists( map ) )
+	{
+		ADMP( va( "%s %s", QQ( N_("^3changedevmap:^* invalid map name '$1$'") ), map ) );
+		return false;
+	}
+
+	if ( trap_Argc() > 2 )
+	{
+		trap_Argv( 2, layout, sizeof( layout ) );
+
+		if ( !Q_stricmp( layout, S_BUILTIN_LAYOUT ) ||
+		     G_LayoutExists( map, layout ) )
+		{
+			// nothing to do
+		}
+		else
+		{
+			ADMP( va( "%s %s", QQ( N_("^3changedevmap:^* invalid layout name '$1$'") ), layout ) );
+			return false;
+		}
+	}
+
+	admin_log( map );
+	admin_log( layout );
+
+	trap_SendConsoleCommand( Str::Format( "devmap %s %s", Cmd::Escape( map ), Cmd::Escape( layout ) ).c_str() );
+
+	level.restarted = true;
+	G_MapLog_Result( 'M' );
+	G_admin_action( QQ( N_("^3changedevmap:^* map '$2$' started by $1$^* $3t$$4$$5$") ),
```

Comment:
Perhaps add "(with Cheats enabled)" at the end of this line?

Reply:
Done